### PR TITLE
Related search update

### PR DESCRIPTION
**Pull request by asifaminb**

### CODE REVIEW

**Review by devonpis — APPROVED** [review]
asserting `any` type is not a good practice in typescript unless there's no other way around, as it just bypass type checking (which is the reason using typescript), the `any` type should have been pick up by linting (tslint) if set up properly (together with eslint).

would be nice if it can be addressed this in the future.

**Comment by asifaminb** [comment]
> asserting `any` type is not a good practice in typescript unless there's no other way around, as it just bypass type checking (which is the reason using typescript), the `any` type should have been pick up by linting (tslint) if set up properly (together with eslint).
> 
> would be nice if it can be addressed this in the future.

Brower event types are not well documented in Typescript,  I have placed a type based on some analysis.

File: package.json [hunk 1]
```diff
@@ -1,7 +1,10 @@
 {
   "name": "global-search",
   "source": "src/index.html",
-  "browserslist": "> 0.25%, not dead",
+  "browserslist": [
```
Comment (duttonw) [inline package.json:4]: interesting, so which plugins use this? https://github.com/browserslist/browserslist

Comment (duttonw) [reply package.json:4]: https://browserslist.dev/?q=PiAwLjI1JSwgbm90IGRlYWQsIGlPUyA%2BPSAxMg%3D%3D

**Review by duttonw — APPROVED** [review]
side topic: if eval or inline js was removed in the future via CSP ( https://content-security-policy.com/examples/allow-inline-script/ ) 

How would you go about the 'search' button etc.

**Comment by asifaminb** [comment]
> side topic: if eval or inline js was removed in the future via CSP ( https://content-security-policy.com/examples/allow-inline-script/ )
> 
> How would you go about the 'search' button etc.

That inline JS is more of a Framework function. It is not exactly similar to raw HTML with vanilla js because with frameworks that are compiled to just HTML without any additional click events

image

File: src/template/related-search.ts [hunk 2]
```diff
@@ -1,13 +1,40 @@
-import { html } from 'lit-html'
+import { html, render } from 'lit-html'
+import { urlParameterMap } from '../utils/urlParameter'
+import { fetchData } from '../utils/fetchData'
+import { mainTemplate } from './main'
+import { noResultsTemplate } from './no-results'
 
 export function relatedResultsTemplate (contextualNavigation: { categories: any; }) {
+  const onRelatedSearchClick = (e: { preventDefault: () => void; target: { href: any; textContent: any } }) => {
```
Comment (devonpis) [inline src/template/related-search.ts]: probably you could define a type and reuse it in this file:
```
type ClickEvent = { preventDefault: () => void; target: { href?: string; textContent?: string } };
```
or just use `e: InputEvent`

File: package.json (hunk 1 above)
Comment (devonpis) [reply package.json:4]: most bundler like parcel and webpack don't require additional plugins to read and stick to the rule of browserslist, unless need to support ie11

File: src/types/clickEvents.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export interface RelatedSearchClick {
+    preventDefault: () => void; target: { href: any; textContent: any }
```
Comment (devonpis) [inline src/types/clickEvents.ts]: better to use `href?: string; textContent?: string` if you know the value is either string or undefined, again minimise the use of `any`

**Pull request merged**